Makefile

```
@@ -185,7 +185,7 @@ test-ts-compat: $(GEN)/protobuf-test node_modules
 		formatted=$$(echo "$${number}" | sed -r 's/[\.]/_/g'); \
 		dirname=packages/protobuf-test/typescript ; \
 		echo "Testing TypeScript `node_modules/ts$$formatted/bin/tsc --version`" ; \
-		node_modules/ts$$formatted/bin/tsc -p $$dirname/tsconfig.$${formatted}.json --noEmit || exit ; \
+		node_modules/ts$$formatted/bin/tsc -p $$dirname/tsconfig.$${formatted}.json --outDir $$dirname/out/$$formatted || exit ; \
```

We already have the `dist` directory with ephemeral artifacts (this is not published, and will never be).

Can you change this to emit into this directory? Than we don't need to ignore a new one. 

```suggestion
		node_modules/ts$$formatted/bin/tsc -p $$dirname/tsconfig.$${formatted}.json --outDir packages/protobuf-test/dist/$$formatted || exit ; \
```

---

Agreed. I'm going to re-use the `dirname` variable there then though if you don't mind!